src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int ARRAY_INDEX_FOR_DATE = 0;
+    private static final int ARRAY_INDEX_FOR_NAME = 1;
+    private static final int ARRAY_INDEX_FOR_WORKING_HOURS = 2;
+    private static final int ARRAY_INDEX_FOR_INCOME_PER_HOUR = 3;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int salary = 0;
+        StringBuilder salaryInfo = new StringBuilder();
+        for (String name : names) {
+            if (name != null) {
+                for (String datum : data) {
+                    String[] dataEntry = datum.split(" ");
+                    if (name.equals(dataEntry[ARRAY_INDEX_FOR_NAME])
+                            && isDatePass(dataEntry[ARRAY_INDEX_FOR_DATE], dateFrom, dateTo)) {
+
+                        salary += Integer.parseInt(dataEntry[ARRAY_INDEX_FOR_WORKING_HOURS])
+                                * Integer.parseInt(dataEntry[ARRAY_INDEX_FOR_INCOME_PER_HOUR]);
+                    }
+                }
+                StringBuilder stringBuilder = new StringBuilder(name);
+                if (name.equals(names[names.length - 1])) {
+                    salaryInfo.append(stringBuilder.append(" - ").append(salary));
+                } else {
+                    salaryInfo.append(stringBuilder.append(" - ").append(salary))
+                            .append(System.lineSeparator());
+                }
```

Comment:
if you will not add lineSeparator in the end of record but will start new record with it you can get rid of this if/else construction

Reply:
then the line at the beginning will be skipped and the tests fail